lib/plausible/billing/subscription/status.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Plausible.Billing.Subscription.Status do
+  @moduledoc """
+  The subscription statuses are stored in Paddle. They can only be changed
+  through Paddle webhooks, which always send the current subscription status
+  via the payload.
+
+  * `active` - All good with the payments. Can access stats.
+
+  * `past_due` - The payment has failed, but we're trying to charge the customer
+    again. Access to stats is still granted. There will be three retries - after
+    3, 5, and 7 days have passed from the first failure. After a failure on the
+    final retry, the subscription status will change to `paused`. As soon as the
+    customer updates their billing details, Paddle will charge them again, and
+    after a successful payment, the subscription will become `active` again.
+
+  * `paused` - we've tried to charge the customer but all the retries have failed.
+    Stats access restricted. As soon as the customer updates their billing details,
+    Paddle will charge them again, and after a successful payment, the subscription
+    will become `active` again.
+
+  * `deleted` - The customer has triggered the cancel subscription action. Access
+    to stats should be granted for the time the customer has already paid for. If
+    they want to upgrade again, new billing details have to be provided.
+
+  Paddle documentation links for reference:
+
+  * Subscription statuses -
+    https://developer.paddle.com/classic/reference/zg9joji1mzu0mdi2-subscription-status-reference
+
+  * Payment failures -
+    https://developer.paddle.com/classic/guides/zg9joji1mzu0mduy-payment-failures
+  """
+
+  @type status() :: :active | :past_due | :paused | :deleted
+
+  defmacro __using__(_opts) do
+    quote do
+      require Plausible.Billing.Subscription.Status
+      alias Plausible.Billing.Subscription
+    end
+  end
+
+  defmacro active(), do: :active
+  defmacro past_due(), do: :past_due
+  defmacro paused(), do: :paused
+  defmacro deleted(), do: :deleted
+
+  def valid_statuses() do
+    [active(), past_due(), paused(), deleted()]
+  end
```

Comment:
Absolutely nothing wrong with what's here, but there's a potential to make things a bit more DRY within the module 😄 

```suggestion
  @statuses [:active, :past_due, :paused, :deleted]
  
  @type status() :: unquote(Enum.reduce(@features, &{:|, [], [&1, &2]}))

  defmacro __using__(_opts) do
    quote do
      require Plausible.Billing.Subscription.Status
      alias Plausible.Billing.Subscription
    end
  end

  for status <- @statuses do
    defmacro unquote(status)(), do: unquote(status)
  end

  def valid_statuses() do
    @statuses
  end
```

Reply:
I like that suggestion. Done in https://github.com/plausible/analytics/pull/3397/commits/1bb8a39fdbe72f67806b04db40d75601743d984f